dhcptest.d

```diff
@@ -501,7 +524,10 @@ bool receivePackets(Socket socket, bool delegate(DHCPPacket, Address) handler, D
 				return true;
 		}
 		catch (Exception e)
+		{
 			stderr.writefln("Error while parsing packet [%(%02X %)]: %s", receivedData, e.toString());
+			end = Clock.currTime();
```

Comment:
That code is to ensure that the while loop would immediately exit on the next go-around.  In my testing I tried things like throwing an invalid format string to the --print-only option, and when the printpackets function was called I had it throw the same exception as the generatepackets function throws when an invalid format was passed.  The exception thrown in printpackets ended up caught there (the output indicated the error processing the packet) and then it would try to receive another packet, which if none came meant the program hung until "timeout".  For instance, --query (no --wait) would have a timeout of "forever". In the past printpackets wouldn't throw an error, so once the receivepackets exited the first time it never was called again. Options like "--query --print-only 54[bogus]" would cause the exception to be thrown and then it would wait until "forever" passed. I verified this by setting timeout to a shorter value and confirming the program eventually exited. The code now probably isn't needed since I ended up just printing an error from the printpackets function instead of throwing an exception.

I am not experienced with D (I had never heard of it before last week) so I don't know how to correctly throw and catch the exception for the case of a bogus format specifier.  Consider it a to-do upgrade/cleanup on top of my contribution. :)